javascript/javascript1/week4/get-reply.js

```diff
@@ -0,0 +1,138 @@
+    todo.push(activity);
+    return activity + "added to your todo";
+  }
+
+  // the following code is to remove from "todo list"
+  if (inputString.search("Remove") == 0) {
+    let activity = inputString.substring(7, inputString.search("from my todo"));
+    todo.pop(activity);
+    return "Removed " + activity + "from your todo";
+  }
+
+  // the following code is to dispaly the activities todo
+  if (inputString === "What is on my todo?") {
+    let activities = "";
+    for (let i = 0; i < todo.length; i++) {
+      if (i == todo.length - 1) {
+        // this part is to display last activity with and
+        activities += "and " + todo[i];
+      } else {
+        // this part is to display all activities except last with ,
+        activities += todo[i] + ",";
+      }
+    }
+    return activities;
+  }
+
+  // the following code is to dispaly date in human readable format
+  if (inputString === "What day is it today?") {
+    let today = new Date();
+    const monthNames = [
+      "January",
+      "February",
+      "March",
+      "April",
+      "May",
+      "June",
+      "July",
+      "August",
+      "September",
+      "October",
+      "November",
+      "December",
+    ];
+    return (
+      today.getDay() +
+      ". of " +
+      monthNames[today.getMonth()] +
+      " " +
+      today.getFullYear()
+    );
+  }
+  // the following code is to do simple mathematics
+  if (inputString.substring(0, 7) == "what is") {
+    // now extract mathematical expression.
+    let mathematicalExpression = inputString.substring(8);
+    if (mathematicalExpression.search("\\+") != -1) {
+      //because + is a meta character we had to use \\+
+      // get the two numbers from the mathematical expression
+      let number1 = mathematicalExpression.substring(
+        0,
+        mathematicalExpression.search("\\+") - 1
+      );
+      let number2 = mathematicalExpression.substring(
+        mathematicalExpression.search("\\+") + 1
+      );
+      return parseInt(number1) + parseInt(number2);
+    } else if (mathematicalExpression.search("-") != -1) {
+      // get the two numbers from the mathematical expression
+      let number1 = mathematicalExpression.substring(
+        0,
+        mathematicalExpression.search("-") - 1
+      );
+      let number2 = mathematicalExpression.substring(
+        mathematicalExpression.search("-") + 1
+      );
+      return parseInt(number1) - parseInt(number2);
+    } else if (mathematicalExpression.search("\\*") != -1) {
+      // since * is a meta character we had to use \\*
+      // get the two numbers from the mathematical expression
+      let number1 = mathematicalExpression.substring(
+        0,
+        mathematicalExpression.search("\\*") - 1
+      );
+      let number2 = mathematicalExpression.substring(
+        mathematicalExpression.search("\\*") + 1
+      );
+      return parseInt(number1) * parseInt(number2);
+    } else return "not a valid input";
+  }
```

Comment:
This is fine. You can also change `inputString` to array of words then get numbers and operation like this
```js
  const commandWords = inputString.split(' ');
  const number1 = parseInt(commandWords[2]);
  const number2 = parseInt(commandWords[4]);
  const operation = commandWords[3];
  if (number1 && number2) {
    switch (operation) {
      case '+':
        return number1 + number2;
      case '-':
        return number1 - number2;
      case '*':
        return number1 * number2;
      case '/':
        return number1 / number2;
    }
  }
```
Also, you can use `eval()` which Evaluates JavaScript code and executes it.
BUT NOT RECOMMENDED.

also many ways using `regular expressions`  can help